### ISSUE TITLE
linter: no-unused-vars false positive when assign variable to class properties

### ISSUE THREAD

**Issue by 0x-jerry** [oxc-project/oxc#12652]
### What version of Oxlint are you using?

latest

### What command did you run?

online playground

### What does your `.oxlintrc.json` config file look like?

```jsonc
{}
```

### What happened?

[Oxc Playground](https://playground.oxc.rs/#eNo9j0FrwzAMhf+K0WmjOXSDXQI7FEavG9s1F8VRisGRg+S0KyH/vXJDcrIkS997bwYPNUTKDt2nOzbcsI+o6k5ubtg5b1M8HBpeyhfTzZ1eXqGCBPUMMnF59M4Z/6HOMlEFMXDeavVppL25D22KW5cFWfskA9Q9RqWlghFFSQoRY0y3X8qT8PeUNXR0ntjnkExvPR+FbPdKPyjEuk4NUcSfCKt3hZWZUS5kzoD0/fj2YRmCpoiZui+yyIIFb6TNjU8dXegZkBjbSH9pEk8DjrvaEDj0YdPzibOkeDbr5ehK0ia19CuwGDJx1AzLAzkNgok=)

[ESLint Playground](https://eslint.org/play/#eyJ0ZXh0IjoibGV0IGEgPSAwXG5cbmNsYXNzIEEge1xuICBjID0gYSsrXG59XG5cbm5ldyBBKCkiLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVudXNlZC12YXJzIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

```ts
let a = 0

class A {
  c = a++
}

new A()
```

oxlint report error, but ESLint not: 

```
  ! eslint(no-unused-vars): Variable 'a' is assigned a value but never used.
  | Unused variables should start with a '_'.
   ,-[:1:5]
 1 | let a = 0
   :     |
   :     `-- 'a' is declared here
 2 | 
 3 | class A {
 4 |   c = a++
   :       |
   :       `-- it was last assigned here
 5 | }
   `----
  help: Did you mean to use this variable?
```
